modules/firefox/hm.nix

```
@@ -159,12 +161,19 @@ in
     );
     warnings = eachTarget (
       { target, ... }:
+      let
+        missingProfiles = lib.lists.subtractLists (builtins.attrNames
+          config.programs.${target.path}.profiles
+        ) config.stylix.targets.${target.path}.profileNames;
+        warn = !config.stylix.targets.${target.path}.disableWarnings;
+      in
       lib.optional
-        (
-          config.programs.${target.path}.enable
-          && config.stylix.targets.${target.path}.profileNames == [ ]
-        )
-        ''stylix: ${target.path}: `config.stylix.targets.${target.path}.profileNames` is not set. Declare profile names with 'config.stylix.targets.${target.path}.profileNames = [ "<PROFILE_NAME>" ];'.''
+        (config.programs.${target.path}.enable && missingProfiles != [ ] && warn)
```

If you accept the above suggestions:

```suggestion
        (config.programs.${target.path}.enable && missingProfiles != [ ] && config.programs.${target.path}.enableProfileWarnings)
```